packages/graphql/src/schema-model/attribute/model-adapters/AttributeAdapter.ts

```diff
@@ -177,17 +177,24 @@ export class AttributeAdapter {
 
     isAggregableField(): boolean {
         return (
-            !this.typeHelper.isList() && (this.typeHelper.isScalar() || this.typeHelper.isEnum()) && this.isAggregable()
+            !this.typeHelper.isList() &&
+            !this.typeHelper.isID() &&
+            (this.typeHelper.isScalar() || this.typeHelper.isEnum()) &&
+            this.isAggregable()
         );
     }
 
     isAggregationWhereField(): boolean {
-        const isGraphQLBuiltInScalarWithoutBoolean =
-            this.typeHelper.isGraphQLBuiltInScalar() && !this.typeHelper.isBoolean();
-        const isTemporalWithoutDate = this.typeHelper.isTemporal() && !this.typeHelper.isDate();
+        if (
+            this.typeHelper.isList() ||
+            this.typeHelper.isID() ||
+            this.typeHelper.isBoolean() ||
+            this.typeHelper.isDate()
+        ) {
+            return false;
+        }
```

Comment:
This is so much easier to read. We should change the others to be like this! :+1:

Reply:
Thank you!